besu/src/main/java/org/hyperledger/besu/components/BesuComponent.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Hyperledger Besu Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.hyperledger.besu.components;
+
+import org.hyperledger.besu.cli.BesuCommand;
+import org.hyperledger.besu.ethereum.bonsai.cache.CachedMerkleTrieLoader;
+import org.hyperledger.besu.ethereum.bonsai.cache.CachedMerkleTrieLoaderModule;
+import org.hyperledger.besu.metrics.MetricsSystemModule;
+import org.hyperledger.besu.metrics.ObservableMetricsSystem;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import dagger.Component;
+import org.slf4j.Logger;
+
+/** An application context that knows how to provide dependencies based on Dagger setup. */
+@Singleton
+@Component(
+    modules = {
+      BesuCommandModule.class,
+      MetricsSystemModule.class,
+      CachedMerkleTrieLoaderModule.class
```

Comment:
Would love to get more maintainers eyes on this PR, particularly regarding this question, before this is merged and a precedent for how to design modules is set :)

Reply:
Devs should DEFINITELY not assume this establishes a precedent of any kind. The right place to do this could be a judgement call. If something is worth being Daggerized within its submodule (evmtool is an example) then the dagger stuff should live right next to its implementation. In other cases, there could be a module that is only ever useful in a BesuContext (like configuration related things).